AvrDataTypes/main/lib/AvrDataTypes.c

```diff
@@ -88,17 +88,59 @@ char* substring(char* str, uint8_t start, uint8_t end) {
     return buffer;
 }
 
-volatile typeof(uint8_t) PORT[] = {0, 1, 2, 3, 4, 5, 6, 7};
-
 /**
- * @brief Gets the Length of array [PORT*] from its size.
- * 
- * @param PORT[] change the PORT array and re-run this to get the length of the new PORT array
- * @return size_t 
+ * @brief Converts a register to an array logic of bits using utility methods.
  */
-size_t getLength() {
-        return sizeof(PORT) / sizeof(typeof(*PORT));
-}
+namespace Register {
+
+    const static uint8_t PORT[] = {0, 1, 2, 3, 4, 5, 6, 7};
+    const static uint8_t HIGH = 0b01;
+    const static uint8_t LOW = 0b00;
+
+    struct Register8 {
+
+        /**
+         * @brief Gets the Length of array [PORT*] from its size.
+         * 
+         * @param PORT[] change the PORT array and re-run this to get the length of the new PORT array
+         * @return size_t 
+         */
+        size_t getLength() {
+            return sizeof(PORT) / sizeof(typeof(*PORT));
+        }
+
+        /**
+         * @brief Digitally writes a [STATE] to a [PIN] in a particular [PORT].
+         * 
+         * @param PORT the register
+         * @param PIN the pin
+         * @param STATE the STATE, either [Regiter::HIGH] or [Register::LOW]
+         */
+        void digitalWrite(volatile uint8_t& PORT, const uint8_t& PIN, const uint8_t& STATE) {
+            uint8_t* BIT = (uint8_t*) calloc(1, sizeof(uint8_t*));
+            *BIT = (HIGH << PIN);
+            if (STATE >= HIGH) {
+                PORT |= *BIT;
+            } else {
+                PORT &= ~(*BIT);
+            }
+            free(BIT);
+        }
+
+        /**
+         * @brief Digitally reads a [PIN] in a [PORT].
+         * 
+         * @param PORT the register
+         * @param PIN the pin to read from
+         * @return const uint8_t a new unsigned char decimal number representing the current state or zero
+         */
+        const uint8_t digitalRead(volatile uint8_t& PORT, const uint8_t& PIN) {
```

Comment:
Here is how `Register::Register8::digitalRead(...)` works: 
1) 2 inputs are involved: `PORT` and ``PIN`.
2) Get the `BIT` which is the binary location of the `PIN` by left shifting  1 (HIGH) by PIN (location of the bit).
3) Read = PORT * BIT, as if (BIT_IN_PORT = 1) then the result will be HIGH at that bit, giving out a unique encoding value, for example `PB5 = 0b00010000` which is the same as 32 in decimal, otherwise it will be low.
4) Libraries can map those unique encoded decimal values for different types of pins and use them to compare if that pin is 1 (HIGH) or 0 (LOW).